src/scene/vcModel.cpp

```diff
@@ -758,10 +689,33 @@ udDouble3 vcModel::GetLocalSpacePivot()
 
 udDouble4x4 vcModel::GetWorldSpaceMatrix()
 {
-  return m_sceneMatrix;
+  return udDouble4x4::translation(m_pivot) * udDouble4x4::rotationQuat(m_orientationCurrent, m_positionCurrent) * udDouble4x4::scaleUniform(m_scaleCurrent.x) * udDouble4x4::translation(-m_pivot);
 }
 
 vcGizmoAllowedControls vcModel::GetAllowedControls()
 {
   return vcGAC_AllUniform;
 }
+
+void vcModel::SetTransforms(const udDouble4x4 &matrix)
+{
+  (udDouble4x4::translation(-m_pivot) * matrix * udDouble4x4::translation(m_pivot)).extractTransforms(m_positionCurrent, m_scaleCurrent, m_orientationCurrent);
```

Comment:
Splitting this over 2 lines will make it more clear what it does.

Reply:
Ok